src/Jackett.Common/Indexers/AwesomeHD.cs

```diff
@@ -0,0 +1,188 @@
+            TorznabCaps.SupportsImdbTVSearch = true;
+
+            AddCategoryMapping(1, TorznabCatType.MoviesHD);
+            AddCategoryMapping(2, TorznabCatType.TVHD);
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            if (configData.Passkey.Value.Length != 32)
+                throw new Exception("Invalid Passkey configured. Expected length: 32");
+
+            var releases = await PerformQuery(new TorznabQuery());
+            await ConfigureIfOK(string.Empty, releases.Any(),
+                                () => throw new Exception("Could not find release from this URL."));
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var releases = new List<ReleaseInfo>();
+            var passkey = configData.Passkey.Value;
+
+            var qc = new NameValueCollection
+            {
+                {"passkey", passkey}
+            };
+
+            if (!string.IsNullOrWhiteSpace(query.ImdbID))
+            {
+                qc.Add("action", "imdbsearch");
+                qc.Add("imdb", query.ImdbID);
+            }
+            else if (!string.IsNullOrWhiteSpace(query.GetQueryString()))
+            {
+                qc.Add("action", "titlesearch");
+                qc.Add("title", query.SearchTerm); // not use query.GetQueryString(), see the season code below
+            }
+            else
+                qc.Add("action", "latestmovies");
+
+            var searchUrl = SiteLink + SearchUrl + "?" + qc.GetQueryString();
+            var results = await RequestStringWithCookies(searchUrl);
+            if (string.IsNullOrWhiteSpace(results.Content))
+                throw new Exception("Empty response. Please, check the Passkey.");
+
+            try
+            {
+                var doc = XDocument.Parse(results.Content);
+
+                var torrents = doc.Descendants("torrent").ToList();
+                if (!torrents.Any())
+                    return releases; // no results
+
+                var authkey = doc.Descendants("authkey").First().Value;
+
+                foreach (var torrent in torrents)
+                {
+                    var torrentName = torrent.FirstValue("name").Trim();
+
+                    // the field <type> is always Movie, so we have to guess if it's a tv series
+                    var isSerie = torrentName.Contains(": Season ");
+                    if (isSerie)
+                        torrentName = torrentName.Replace(": Season ", " S");
+
+                    // if the category is not in the search categories, skip
+                    var cat = new List<int> {isSerie ? TorznabCatType.TVHD.ID : TorznabCatType.MoviesHD.ID};
+                    if (query.Categories.Any() && !query.Categories.Intersect(cat).Any())
```

Comment:
Combine sequential checks:
```suggestion
                    if (!query.Categories.Intersect(cat).Any())
```

Reply:
Sorry but the logic is different.
If `query.Categories` is empty, that means all categories and I don't want to remove results. With your code the intersection will be empty and I remove all results.

Reply:
Gotcha. Then the only other thing here is that maybe instead of building a list for a single value and calling intersect, we should just call `.Conatins()` from LINQ? Works the same way as `List.Contains()` but works on all IEnumerable types, including arrays.